data/finset.lean

```diff
@@ -978,6 +978,72 @@ by haveI := classical.prop_decidable;
 
 end fold
 
+/- max and min of finite sets -/
+section max_min
+variables [decidable_linear_order α]
+
+def max (a : α) : finset α → α :=
```

Comment:
The problem with `max` is that it clashes with `_root_.max` (as you see in the definition).
I think this should be `protected`. Similar for `min`.

Reply:
I have no issue with that. Should the names like `le_max_left` also be protected? Or should they change? (See the other uses of `_root_.`.)